.github/workflows/livetest.yml

```diff
@@ -0,0 +1,149 @@
+name: Live test cockpit from PRs
+# Triggered when a repository collaborator comments /livetest on a PR
+# This is currently configured to run on cockpit-project org.
+# If your project lives somewhere else than the github cockpit-project organization,
+# you need to set up your own relay host, add a secret with the private SSH key, and
+# adjust the RELAY_HOST env variable and SINK_SSHRELAY_PRIVATE secret name.
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  pr-info:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Query author repository permissions
+        uses: octokit/request-action@v2.x
+        id: user_permission
+        with:
+          route: GET /repos/${{ github.repository }}/collaborators/${{ github.event.sender.login }}/permission
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # restrict /livetest to users with admin or write permission for the repository
+      # see https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-repository-permissions-for-a-user
+      # store output if user is allowed in allowed_user job output so it has to be checked in downstream job
+      - name: Check if user does have correct permissions
+        if: contains('admin write', fromJson(steps.user_permission.outputs.data).permission)
+        id: check_user_perm
+        run: |
+          echo "User '${{ github.event.sender.login }}' has permission '${{ fromJson(steps.user_permission.outputs.data).permission }}' allowed values: 'admin', 'write'"
+          echo "::set-output name=allowed_user::true"
+
+    outputs:
+      allowed_user: ${{ steps.check_user_perm.outputs.allowed_user }}
+
+  live-test:
+    needs: pr-info
+    if: needs.pr-info.outputs.allowed_user == 'true' && github.event.issue.pull_request && startsWith(github.event.comment.body, '/livetest')
+    runs-on: ubuntu-20.04
+    env:
+      RELAY_HOST: logs.cockpit-project.org
+    steps:
+      - name: Set up configuration and secrets
+        run: |
+          echo '${{ secrets.GITHUB_TOKEN }}' > ~/.config/github-token
+
+      - uses: actions/github-script@v3
+        id: get-pr
+        with:
+          script: |
+            const request = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            }
+            core.info(`Getting PR #${request.pull_number} from ${request.owner}/${request.repo}`)
+            try {
+              const result = await github.pulls.get(request)
+              return result.data
+            } catch (err) {
+              core.setFailed(`Request failed with error ${err}`)
+            }
+
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ fromJSON(steps.get-pr.outputs.result).head.repo.full_name }}
+          fetch-depth: 0
+          ref: ${{ fromJSON(steps.get-pr.outputs.result).head.ref }} # or .head.ref for branch name
```

Comment:
That is an odd comment. starter-kit does not have that, I guess it needs to be retrofitted? Otherwise it would always checkout master, not the PR, right?

Reply:
Right, this needs to be fixed for starter-kit and podman, they are both wrong now.

Reply:
Removed comment.